src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioMacDocumentInfoFactory.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    /// <summary>
+    /// This class is VisualStudio for Mac specific to enable creating an empty document info without having IVT access to Roslyn's types.
+    /// </summary>
+    [Shared]
+    [Export(typeof(VisualStudioMacDocumentInfoFactory))]
+    internal class DefaultVisualStudioMacDocumentInfoFactory : VisualStudioMacDocumentInfoFactory
```

Comment:
This factory wont be needed once https://github.com/dotnet/roslyn/pull/39673 flows. For now this is a sufficient workaround.